app/Providers/FastlyProvider.php

```diff
@@ -64,8 +64,87 @@ public function boot()
         });
 
         //deleted
+         Redirect::deleted(function($redirect) {
+            Log::info(sprintf('Deleting Fastly redirect: %s => %s (%d)',
+                $redirect->path, $redirect->target, $redirect->http_status));
+
+
+            // Here's the hacky part....
+            $fastly_key = '30342701a122f58e935c9fe1024b8bc8';
+            $fastly_table_redirects = 'isbuug15ZlBduv6XWk0VU';
+            $fastly_table_redirect_types = '5v35JNHgaK2qFG5fNacSPO';
+
+            // Step one: Remove from 'redirects' table.
+            $ch = curl_init();
+            curl_setopt_array($ch, array(
+                CURLOPT_RETURNTRANSFER => 1,
+                CURLOPT_CUSTOMREQUEST => "DELETE",
+                CURLOPT_HTTPHEADER => array(sprintf('Fastly-Key: %s', $fastly_key)),
+                CURLOPT_URL => 'https://api.fastly.com/service/3uoN6UPm3Byl5RAkYuhTTk/dictionary/' . $fastly_table_redirects . '/item/' . $redirect->path,
+            ));
+            $response = curl_exec($ch);
```

Comment:
Definitely not necessary, but this might read nicer using a shared Guzzle client:

```php
$this->fastly = new Client([
  'base_url' => 'https://api.fastly.com/',
  'defaults' => [
    'headers' => [
      'Fastly-Key' => $fastly_key,
    ]
  ]
])
```

And then make calls like this later on: 
```php
$this->fastly->delete('service/3uoN6UPm3Byl5RAkYuhTTk/dictionary/' . $fastly_table_redirects . '/item/' . $redirect->path);
```